libgadget/domain.c

```diff
@@ -417,7 +406,7 @@ domain_balance(void)
 }
 
 static int
-domain_check_memory_bound(const int print_details, int64_t *TopLeafWork, int64_t *TopLeafCount)
+domain_check_memory_bound(const int print_details, int64_t *TopLeafWork, int64_t *TopLeafCount, const Domain * domain)
```

Comment:
Is there a consistent policy when to put Domain* as the first argument, when to put it as the last?

Reply:
I put it last when it was const and used as input userdata, first when it was being created and built and behaving like a self pointer. I can make it consistently first if you like